assets/js/modules/adsense/components/common/AdBlockingRecoveryToggle.test.js

```diff
@@ -0,0 +1,111 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import AdBlockingRecoveryToggle from './AdBlockingRecoveryToggle';
+import { render, provideModules } from '../../../../../../tests/js/test-utils';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED,
+	MODULES_ADSENSE,
+} from '../../datastore/constants';
+import { ACCOUNT_STATUS_READY, SITE_STATUS_READY } from '../../util';
+import { enabledFeatures } from '../../../../features';
+
+describe( 'AdBlockingRecoveryToggle', () => {
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: false,
+		accountStatus: ACCOUNT_STATUS_READY,
+		siteStatus: SITE_STATUS_READY,
+		adBlockingRecoverySetupStatus: '',
+	};
+
+	it.each( [
+		[ 'the Ad blocker detection feature flag is not enabled', false ],
+		[ 'Ad blocker recovery setup status is empty', true ],
+	] )(
+		'should not render the Ad blocking recovery toggle when %s',
+		( testName, adBlockerDetectionEnabled ) => {
+			const { container } = render( <AdBlockingRecoveryToggle />, {
+				setupRegistry: ( registry ) => {
+					if ( adBlockerDetectionEnabled ) {
+						enabledFeatures.add( 'adBlockerDetection' );
+					}
+					provideModules( registry, [
+						{
+							slug: 'adsense',
+							active: true,
+							connected: true,
+						},
+					] );
+					registry
+						.dispatch( MODULES_ADSENSE )
+						.receiveGetSettings( validSettings );
+				},
+			} );
+
+			expect(
+				container.querySelector(
+					'.googlesitekit-settings-module__ad-blocking-recovery-toggles'
+				)
+			).toBeNull();
+
+			expect( container.textContent ).not.toContain(
+				'Ad blocking recovery'
+			);
+		}
+	);
+
+	it( 'should render the Ad blocking recovery toggle when the conditions are met', () => {
+		const { container } = render( <AdBlockingRecoveryToggle />, {
+			setupRegistry: ( registry ) => {
+				enabledFeatures.add( 'adBlockerDetection' );
+				provideModules( registry, [
+					{
+						slug: 'adsense',
+						active: true,
+						connected: true,
+					},
+				] );
+				registry.dispatch( MODULES_ADSENSE ).receiveGetSettings( {
+					...validSettings,
+					adBlockingRecoverySetupStatus:
+						AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED,
+					useAdBlockerDetectionSnippet: true,
+					useAdBlockerDetectionErrorSnippet: true,
+				} );
+			},
+		} );
+
+		expect(
+			container.querySelector(
+				'.googlesitekit-settings-module__ad-blocking-recovery-toggles'
+			)
+		).not.toBeNull();
+
+		expect( container.textContent ).toContain( 'Ad blocking recovery' );
+		expect(
+			container.querySelector(
+				'.googlesitekit-settings-module__meta-item'
+			)
+		).toHaveTextContent( 'Place ad blocking recovery tag' );
+	} );
```

Comment:
Please add couple more test here similar to the stories. Tests where it asserts\ that disabling Ad Blocking Recovery tag hides the error protection toggle, the behavior of notice on toggle change and on mount.